pkg/diag/validator/config_connector.go

```diff
@@ -0,0 +1,168 @@
+		} else {
+			// config connector status doesn't always correctly parse to failed, but shows InProgress with an error message
+			status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_FAILED, result.Message)
+		}
+
+	case kstatus.FailedStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_FAILED, result.Message)
+	case kstatus.TerminatingStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_TERMINATING, result.Message)
+	case kstatus.NotFoundStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_NOT_FOUND, result.Message)
+	case kstatus.UnknownStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_UNKNOWN, result.Message)
+	}
+	return status
+}
+func processResourceEvents(e corev1.EventInterface, res unstructured.Unstructured, rs *configConnectorResourceStatus) {
+	log.Entry(context.TODO()).Debugf("Fetching events for config connector resource %q", res.GetName())
+	// Get pod events.
+	scheme := runtime.NewScheme()
+	scheme.AddKnownTypes(v1.SchemeGroupVersion, &res)
+	events, err := e.Search(scheme, &res)
+	if err != nil {
+		log.Entry(context.TODO()).Debugf("Could not fetch events for resource %q due to %v", res.GetName(), err)
+		return
+	}
+	// find the latest event.
+	var recentEvent *v1.Event
+	for _, e := range events.Items {
+		event := e.DeepCopy()
+		if recentEvent == nil || recentEvent.LastTimestamp.Before(&event.LastTimestamp) {
+			recentEvent = event
+		}
+	}
+	if recentEvent == nil || recentEvent.Type == v1.EventTypeNormal {
+		return
+	}
+	// TODO: Add unique error codes for reasons
+	rs.updateAE(
+		proto.StatusCode_STATUSCHECK_UNKNOWN_EVENT,
+		fmt.Sprintf("%s: %s", recentEvent.Reason, recentEvent.Message),
+	)
+}
+
+type configConnectorResourceStatus struct {
+	name      string
+	namespace string
+	result    kstatus.Result
+	ae        proto.ActionableErr
+}
+
+func (s *configConnectorResourceStatus) updateAE(errCode proto.StatusCode, msg string) {
+	s.ae.ErrCode = errCode
+	s.ae.Message = msg
+}
+
+type configConnectorSelector struct {
+	client    kubernetes.Interface
+	dynClient dynamic.Interface
+	uResource unstructured.Unstructured
+}
+
+func NewConfigConnectorSelector(client kubernetes.Interface, dynClient dynamic.Interface, uResource unstructured.Unstructured) CustomResourceSelector {
+	return &configConnectorSelector{client: client, dynClient: dynClient, uResource: uResource}
+}
+
+func (c *configConnectorSelector) Select(ctx context.Context, namespace string, opts metav1.ListOptions) (*unstructured.UnstructuredList, error) {
+	_, r, err := util.GroupVersionResource(c.client.Discovery(), c.uResource.GroupVersionKind())
+	if err != nil {
+		return nil, fmt.Errorf("failed to query config connector resources: %w", err)
+	}
+	resList, err := c.dynClient.Resource(r).Namespace(namespace).List(ctx, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query config connector resources: %w", err)
+	}
+	return resList, nil
+}
```

Comment:
yes, it's to fetch the latest version. Also we poll on this selector so that if there are any new resources of this type that show up later them they are checked on that iteration (as against just fetching the list once and checking the status of known resources individually).